cpp/src/arrow/buffer-test.cc

```diff
@@ -272,7 +273,10 @@ TYPED_TEST(TypedTestBuffer, ResizeOOM) {
   TypeParam buf;
   ASSERT_OK(AllocateResizableBuffer(0, &buf));
   ASSERT_OK(buf->Resize(100));
-  int64_t to_alloc = std::numeric_limits<int64_t>::max();
+  int64_t to_alloc = std::min<uint64_t>(std::numeric_limits<int64_t>::max(),
```

Comment:
`min<int64_t>` and not uint

Reply:
No. The value of `std::numeric_limits<size_t>::max()` can be as big as `std::numeric_limits<uint64_t>::max()` which overflows `int64_t`.